integration-box/yahoo-dmp/scripts/put_userlist.py

```diff
@@ -1,18 +1,43 @@
 import os
 import sys
 import requests
+from logging import DEBUG, INFO, StreamHandler, getLogger
+import gzip
 
-os.system(f"{sys.executable} -m pip install -U pytd==1.0.0 td-client")
+os.system(f"{sys.executable} -m pip install -U pytd==1.3.0 td-client")
 import pytd
 import pandas as pd
 
+logger = getLogger(__name__)
+handler = StreamHandler()
+handler.setLevel(INFO)
+logger.setLevel(INFO)
+logger.addHandler(handler)
+logger.propagate = False
+
+
 def upload(sqlfile, database, presigned_url):
   with open(sqlfile) as f:
     querytxt = f.read()
 
   client = pytd.Client(apikey=os.getenv('td_apikey'), database=database)
   res = client.query(querytxt)
   df = pd.DataFrame(**res)
-  csv = df.to_csv(index=False)
+  csv = df.to_csv(header=False, index=False)
+
+  print('---- user list as first 10 lines----')
+  print('\n'.join(csv.splitlines()[:10]))
+  print('---- Total number of IDs = ' + str(len(csv.splitlines())) + '----')
+
+  res = requests.put(
+          presigned_url,
+          data=gzip.compress(bytes(csv, 'utf-8')),
```

Comment:
JFYI, `DataFrame.to_csv` has `compression` option https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html

Reply:
Thanks! I'll replace it using to_csv compression option on next PR. (I'm preparing to support another Yahoo! DMP API)

Reply:
@chezou 
I just remembered that to_csv(compression='gzip') is not working likely. 
I think in case of serializing to file as 'to_csv(path_or_buf="filepath", compression='gzip')' it's working well. But trying to handle without path_or_buf parameter as 'to_csv(compression='gzip')', compression='gzip' parameter seems be ignored.
(to_csv returned just string object instead of gzip bytes actually)

https://stackoverflow.com/questions/37193157/apply-gzip-compression-to-a-csv-in-python-pandas

The behavior is defined around here.
https://github.com/pandas-dev/pandas/blob/v1.0.5/pandas/io/formats/csvs.py#L157

This is why I gave up to use to_csv with compression='gzip' parameter. If you have any idea, please give me a advice!

Reply:
Make sense. In that case, we might be able to use Tempfile, but it's more straightforward to keep as is, depending on the data size.